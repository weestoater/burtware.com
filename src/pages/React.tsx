import React from "react";
import ReactHooksPost from "../components/react/ReactHooksPost";
import RandomUsersPost from "../components/react/RandomUserPost";

export default function ReactPage() {
  return (
    <div className="row">
      <RandomUsersPost />
      <ReactHooksPost />
      <div className="col-lg-3 col-md-6 mb-4">
        <div className="card">
          <div className="card-header">
            <i className="bi bi-filetype-jsx" />
            <span className="pl-4">my experiences</span>
          </div>
          <div className="card-body">
            <p>
              There are as many ways to do Single Page Applications (SPAs) as
              there are developers with opinions. Within my work we have many
              instances of AngularJS, Angular (4,5,6,7,9,11,12) and ReactJS.
            </p>
            <p>
              My experience with ReactJS began as pretty limited, but with a
              strong following wind, a few cups of coffee and a Pluralsight
              subscription â€“ I resolved that in no small measure.</p>
            <p>
              Before moving recently to pure UX team, I worked on a growing React platform which included multiple chart and grid offerings.
            </p>
            <p>
              I am now trying to get my head around the worlds of <code>typescript</code>; micro-services and AWS with React.
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}
