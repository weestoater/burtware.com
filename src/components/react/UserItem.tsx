import React from "react";

function UserItem(props: any) {

  const user = props;

  return (
    <>
      <div className="user col-sm-12 mb-3">
        <img src={user.item.picture.large} alt="" className="picture" />
        <p className="name">
          <span className="badge bg-info text-black me-2">{user.item.nat}</span>{" "}
          {user.item.name.title} {user.item.name.first} {user.item.name.last}
        </p>
        <address className="address">
          {user.item.location.street.number} {user.item.location.street.name}{" "}
          <br />
          {user.item.location.city}
          <br />
          {user.item.location.state}, {user.item.location.country}
          <br />
          {user.item.location.postcode}
        </address>
        <p className="contact">
          <i className="bi bi-phone" aria-hidden="true" />{" "}
          <a href={`tel:${user.item.cell}`}>{user.item.cell}</a>
          <br />
          <i className="bi bi-telephone" aria-hidden="true" />{" "}
          <a href={`tel:${user.item.phone}`}>{user.item.phone}</a>
          <br />
          <i className="bi bi-envelope" aria-hidden="true" />{" "}
          <a href={`mailto:${user.item.email}`}>{user.item.email}</a>
        </p>
      </div>
    </>
  );
}

export default UserItem;
