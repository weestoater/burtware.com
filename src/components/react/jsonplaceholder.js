import React, { useState, useEffect } from 'react';

export const JsonPlaceholder = () => {

    // const fetchData = fetch('https://geodb-cities-graphql.p.rapidapi.com', {
    //     method: 'POST',
    //     headers: {'Content-Type': 'application/json'},
    //     body: JSON.stringify({
    //         query:`{
    //             countries {
    //                 name
    //                 capital
    //                 currency
    //             }
    //         }`
    //     })
    // })
    // .then(res => res.json())
    // .then(res => console.log(res.data));

    // fetchData();

    return (
        <div className="card">
            <div className="card-header">FETCH DATA</div>
            <div className="card-body">
                body    
            </div> 
        </div>
    );
}