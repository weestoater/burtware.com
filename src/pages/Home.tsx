import React from "react";
import PostList from "../components/blogpostlist";

export default function Home() {
  return (
    <div className="row">
        <div className="col-sm-12">
            <h1 className="visually-hidden">Home</h1>
        </div>
        <div className="col-lg-6 mb-4">
          <div className="row">
            <PostList />
          </div>
        </div>
        <div className="col-lg-6 mb-4">
            <iframe src="https://www.youtube.com/embed/9zEhX_Gxbkc"
            title="YouTube video player"></iframe>
        </div>        
    </div>
  );
}
