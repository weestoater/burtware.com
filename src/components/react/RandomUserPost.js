import React, {useState, useEffect} from 'react';
import getApi from '../../adapters/commonApi';
import Spinner from '../global/Spinner';
import UserItem from './UserItem';

export default function RandomUsersPost() {

    const [users, setUsers] = useState({});
    const apiUrl = "https://randomuser.me/api/?results=3";

    useEffect(() => {
        getApi(apiUrl).then(setUsers);
      }, []);

    return (
        <div className="col-lg-4 col-md-6 mb-4">
            <div className="card">
                <div className="card-header">
                    <i className="bi bi-filetype-jsx" />
                    <span className="pl-4">Random Users API</span>
                </div>
                <div className="card-body users">
                {users.results ? (
                    users.results.map((user, key) => <UserItem key={key} item={user} /> )
                ) : ( 
                <Spinner /> 
                )}
                </div>
            </div>
        </div>
    );
}
