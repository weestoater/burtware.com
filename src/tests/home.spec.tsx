import React from "react";
import { render, screen } from "@testing-library/react";
import Home from "../pages/Home";

describe('Testing Home page', () => {

    it('can render Homepage', () => {
        render(<Home />);
        const welcome = screen.getByText(/welcome/i);
        expect(welcome).toBeInTheDocument();
    });

    it('can render HomePage', () => {
        const {container} = render(<Home />)
        expect(container).toMatchSnapshot();
    });
});