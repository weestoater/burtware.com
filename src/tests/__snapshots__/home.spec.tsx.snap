// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home page can render HomePage 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-sm-12"
    >
      <h1
        class="white"
      >
        Welcome to 
        <code>
          burtware
        </code>
      </h1>
    </div>
    <div
      class="col-lg-3 mb-2"
    >
      <h2
        class="white"
      >
        Car Videos
      </h2>
      <figure>
        <iframe
          src="https://www.youtube.com/embed/i0Z1VKWf3RM"
          title="YouTube video player 1"
        />
        <figcaption>
          Range Rover Sport vs spillway
        </figcaption>
      </figure>
      <figure>
        <iframe
          src="https://www.youtube.com/embed/R6ygiL366TE"
          title="YouTube video player 2"
        />
        <figcaption>
          Guy Martin in highly modified Land Rover
        </figcaption>
      </figure>
    </div>
    <div
      class="col-lg-3 mb-2"
    >
      <h2
        class="white"
      >
        Music Videos
      </h2>
      <figure>
        <iframe
          src="https://www.youtube.com/embed/iTKJ_itifQg"
          title="YouTube video player 3"
        />
        <figcaption>
          depeche Mode - I Feel You
        </figcaption>
      </figure>
      <figure>
        <iframe
          src="https://www.youtube.com/embed/xy37XdAwH_A"
          title="YouTube video player 4"
        />
        <figcaption>
          Lindemann - Home Sweet Home (live)
        </figcaption>
      </figure>
    </div>
    <div
      class="col-lg-6 mb-2"
    >
      <h2>
        Â 
      </h2>
      <div
        class="card"
      >
        <div
          class="card-header"
        >
          <i
            class="bi bi-wrench-adjustable-circle"
          />
           Ethos
        </div>
        <div
          class="card-body"
        >
          <p>
            I started this space as an online area for me to experiment with ReactJS and data 'patterns' on AWS.  I've not got any desire to add a database or other server-side mechanisms - instead I want to concentrate purely on the front end aspects of rendering 
            <code>
              JSON
            </code>
             payloads. 
          </p>
          <p>
            There isn't any roadmap of what I will build, but I am iterating around the football details, using the Motherwell F.C. results as my data source. I am 'hand-rolling' the data into two 
            <code>
              .json
            </code>
             files: mfc-goals & mfc-matches, then creating a number of components to render the results onto the page.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
